File: codes/Sem1/03_root.py
"""
Role Number: xxxx
Description: Roots of a quadratic equation ax^2+bx+c=0
Author     : AKB
"""

import math, time
root1 = root2 = realp = imagp = discr = 0.0
a,b,c = input('Enter the values of a, b & c: ')
start_time = time.time()

# If a or b or c is 0, it is not a quadratic equation, hence exit 
if a==0: 
   print 'Error: Roots cannot be determined'
   exit(1)
else:
    discr = math.pow(b,2) - 4.0*a*c;
    if discr < 0:
        print 'Imaginary roots\n'
        realp = -b/(2.0*a);
        imagp = math.sqrt(abs(discr))/(2.0*a);
        print 'Root1 = ', realp, '+ i', imagp, '\n'
        print 'Root2 = ', realp, '- i', imagp, '\n'    
    elif discr == 0:
        print 'Roots are real and equal\n'
        root1 = -b/(2.0*a)
        print 'Root1 = Root2 = ', root1, '\n'
    else:
        print 'Roots are real and distinct\n'
        root1 = (-b + math.sqrt(discr))/(2.0*a); 
        root2 = (-b - math.sqrt(discr))/(2.0*a); 
        print 'Root1 = ', root1
        print 'Root2 = ', root2
    
exec_time = time.time() - start_time
print 'Execution time = ', exec_time, ' seconds'

"""
Results 
Try say (1,2,1), (1,4,1), (2,3,4)
"""

File: codes/03_root.c
/* Regn.No.:xxxx
   Description: Root of a quadratic equation ax^2+bx+c=0 
   Author: AKB */

#include<stdio.h>
#include<stdlib.h>
#include<math.h>

int main(){

  /* Type Declaration */ 
  float a, b, c, root1, root2;
  float realp, imagp, discr;
  printf("Enter the values of a, b & c\n");
  scanf("%f %f %f",&a, &b, &c);

  /* If a or b or c is 0, it is not a quadratic equation, so break */
  if(a==0 || b==0 || c==0){
    printf("Error: Roots cannot be determined");
    exit(1);
  }
  else{
    discr = pow(b,2) - 4.0*a*c;
    if(discr < 0){
      printf("Imaginary roots\n");
      realp = -b/(2.0*a);
      imagp = sqrt(abs(discr))/(2.0*a);
      printf("Root1 = %f + i%f\n",realp,imagp);
      printf("Root2 = %f - i%f\n",realp,imagp);
    }
    else if (discr == 0){
      printf("Roots are real and equal\n");
      root1 = -b/(2.0*a); 
      printf("Root1 = %f\n",root1);
      printf("Root2 = %f\n",root1);
    }
    else if (discr > 0){ 
      printf("Roots are real and distinct\n");
      root1 = (-b + sqrt(discr))/(2.0*a); 
      root2 = (-b - sqrt(discr))/(2.0*a); 
      printf("Root1 = %f\n",root1);
      printf("Root2 = %f\n",root2);
    }
  }
  return 0;
}

/* Results */
